src/Microsoft.AspNetCore.SignalR.Core/HubEndPoint.cs

```diff
@@ -450,30 +479,34 @@ private bool IsChannel(Type type, out Type payloadType)
             {
                 await SendMessageAsync(connection, CompletionMessage.WithError(invocationId, ex.Message));
             }
+            finally
+            {
+                if (connection.ActiveRequests.TryRemove(invocationId, out var cts))
```

Comment:
Maybe you can add a method on connection to cancel the invocation - you now do it from two places...?